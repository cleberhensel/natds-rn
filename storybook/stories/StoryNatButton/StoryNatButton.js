import React from 'react';
import {
  ScrollView,
  SafeAreaView,
  Dimensions,
  Alert,
} from 'react-native';
import {
  NatContainer,
  NatText,
  NatButton,
  NatSpace,
} from '../../../lib';

const styles = {
  defaultScreen: {
    flex: 1,
    width: Dimensions.get('window').width,
  },
};

export const StoryNatButton = () => (
    <SafeAreaView style={styles.defaultScreen}>
      <NatContainer>
        <ScrollView>
          <NatText value={'Buttons - Social Selling'} type="h6" />
          <NatSpace />
          <NatButton
            type="default"
            label="default"
            onPress={() => Alert.alert('press button')}
          />
          <NatSpace />
          <NatButton
            type="short"
            label="short"
            onPress={() => Alert.alert('press button')}
          />
          <NatButton
            type="card"
            label="card"
            onPress={() => Alert.alert('press button')}
          />
        </ScrollView>
      </NatContainer>
    </SafeAreaView>
);
