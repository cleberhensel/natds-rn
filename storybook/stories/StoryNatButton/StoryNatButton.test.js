import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components/native';

import { StoryNatButton } from './StoryNatButton';

describe('StoryNatButton component', () => {
  it('Should render StoryNatButton component ', () => {
    const button = renderer.create(
      <StoryNatButton />,
    ).toJSON();

    expect(button).toMatchSnapshot();
  });
});
