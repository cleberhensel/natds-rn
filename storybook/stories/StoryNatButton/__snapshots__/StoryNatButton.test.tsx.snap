// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryNatButton component Should render StoryNatButton component : StoryNatButton default render 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
      "width": 750,
    }
  }
>
  <View
    marginBottom="none"
    marginLeft="none"
    marginRight="none"
    marginTop="none"
    paddingBottom="small"
    paddingLeft="small"
    paddingRight="small"
    paddingTop="small"
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        },
      ]
    }
  >
    <RCTScrollView>
      <View>
        <View
          margin={24}
          style={
            Array [
              Object {
                "marginBottom": 24,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 0,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
              },
            ]
          }
        >
          <Text
            color="#333333"
            style={
              Array [
                Object {
                  "color": "#333333",
                  "fontSize": 20,
                  "fontWeight": "500",
                  "lineHeight": 30,
                },
              ]
            }
            type="h6"
          >
            Buttons - Social Selling
          </Text>
        </View>
        <View
          backgroundColor="transparent"
          marginBottom="small"
          marginLeft="none"
          marginRight="none"
          marginTop="none"
          paddingBottom="none"
          paddingLeft="none"
          paddingRight="none"
          paddingTop="none"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "marginBottom": 16,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 0,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
              },
            ]
          }
        />
        <View
          accessibilityRole="button"
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F4AB34",
              "borderColor": "transparent",
              "borderRadius": 6,
              "borderWidth": 0,
              "flexDirection": "row",
              "height": 48,
              "justifyContent": "center",
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <Text
            accessibilityLabel="default"
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "#FFFFFF",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "bold",
                  "letterSpacing": 1,
                },
              ]
            }
            type="default"
          >
            DEFAULT
          </Text>
        </View>
        <View
          backgroundColor="transparent"
          marginBottom="small"
          marginLeft="none"
          marginRight="none"
          marginTop="none"
          paddingBottom="none"
          paddingLeft="none"
          paddingRight="none"
          paddingTop="none"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "marginBottom": 16,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 0,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
              },
            ]
          }
        />
        <View
          accessibilityRole="button"
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderColor": "transparent",
              "borderRadius": 6,
              "borderWidth": 0,
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "center",
              "opacity": 1,
              "width": 328,
            }
          }
        >
          <Text
            accessibilityLabel="short"
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "#F4AB34",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "bold",
                  "letterSpacing": 1,
                },
              ]
            }
            type="short"
          >
            SHORT
          </Text>
        </View>
        <View
          accessibilityRole="button"
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F4AB34",
              "borderColor": "transparent",
              "borderRadius": 6,
              "borderWidth": 0,
              "flexDirection": "row",
              "height": 128,
              "justifyContent": "center",
              "opacity": 1,
              "width": 128,
            }
          }
        >
          <Text
            accessibilityLabel="card"
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "#FFFFFF",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "bold",
                  "letterSpacing": 1,
                  "lineHeight": 21,
                  "textAlign": "center",
                  "width": 64,
                },
              ]
            }
            type="card"
          >
            CARD
          </Text>
          <Image
            source={
              Object {
                "testUri": "../../../lib/assets/icons/ic_arrow_left_white.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 30,
                  "width": 30,
                },
              ]
            }
          />
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
